blivet/udev.py

```diff
@@ -41,19 +41,31 @@
 
 def get_device(sysfs_path):
     try:
-        dev = pyudev.Devices.from_sys_path(global_udev, sysfs_path)
+        device = pyudev.Devices.from_sys_path(global_udev, sysfs_path)
     except pyudev.DeviceNotFoundError as e:
         log.error(e)
-        dev = None
+        result = None
+    else:
+        result = dict(device.properties)
+        result["SYS_NAME"] = device.sys_name
+        result["SYS_PATH"] = device.sys_path
 
```

Comment:
I would suggest that you make a little function containing the three lines above, so you can reuse the code here and in `get_devices` below.